File: KC.Backend.Models/GameItems/Hand.cs
namespace KC.Backend.Models.GameItems;

public class Hand
{
    public List<Card> Cards { get; set; } = [];
    public bool DealerOwned { get; init; } = false;
    public bool FromSplit { get; set; } = false;
    public double Bet { get; set; } = 0;
    public bool Finished { get; set; } = false;
}

public record struct HandValue(int NumberValue, bool IsBlackJack, bool IsPair, bool IsSoft);

File: KC.Detector/CardCounting/Models/Hand.py
from Card import Card
from HandValue import HandValue

class Hand:
    cards: list[Card]
    def __init__(self):
        self.cards = []

    def add_card(self, card: Card) -> None:
        self.cards.append(card)

    def remove_card(self, card: Card) -> None:
        self.cards.remove(card)

    def clear(self) -> None:
        self.cards = []

    def get_value(self) -> HandValue:
        return HandValue.from_hand(self)

    def __str__(self):
        return ', '.join(str(card) for card in self.cards) + f" (Value: {self.get_value()})"